### DISCUSSION TITLE
help custom endpoint

### DISCUSSION THREAD

**Discussion by emrahamet** [pocketbase/pocketbase#1363] [Q&A] [answered]
hello i need some help
what I want to do is job tracking system

example /customEndpoint/:id

1. must be accessible without logging in
2. I think there will be a problem during the search because the existing id is too long (15)
3. Is it okay to search by existing id or should I use another one?
4. Is there a security problem by doing this?

thanks

#### apis/custom.go
```go
package apis

import (
	"log"
	"net/http"
	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase/core"
)

const collectionNameOrId = "jobs"

// bindCustomCrudApi registers the /custom api endpoint and
// the corresponding handlers.
func bindCustomCrudApi(app core.App, rg *echo.Group) {
	api := customRecordApi{app: app}

	subGroup := rg.Group(
		"/custom",
		ActivityLogger(app),
		// LoadCollectionContext(app),
	)

	subGroup.GET("", api.cErr)
	subGroup.GET("/:id", api.view)
}

type customRecordApi struct {
	app core.App
}

func (api *customRecordApi) cErr(c echo.Context) error{
	return NewNotFoundError("Error message 1", "Custom: Missing Id")
}


func (api *customRecordApi) view(c echo.Context) error {

	recordId := c.PathParam("id")
	if recordId == "" {
		return NewNotFoundError("Error message 2", "Custom: Missing Id")
	}

	// recordId = recordId[4: len(recordId)-4]

	record, fetchErr := api.app.Dao().FindRecordById(collectionNameOrId, recordId, nil)
	if fetchErr != nil || record == nil {
		// return NewNotFoundError("Not Found..", fetchErr)
		return NewNotFoundError("", fetchErr)
	}

	event := &core.RecordViewEvent{
		HttpContext: c,
		Record: record,
	}

	return api.app.OnRecordViewRequest().Trigger(event, func(e *core.RecordViewEvent) error {
		if err := EnrichRecord(e.HttpContext, api.app.Dao(), e.Record); err != nil && api.app.IsDebug() {
			log.Println(err)
		}

		result := map[string]string{
			"name":    e.Record.GetString("name"),
			"state": e.Record.GetString("state"),
		}
		// return e.HttpContext.JSON(http.StatusOK, e.Record)
		return e.HttpContext.JSON(http.StatusOK, result)
	})
}

```

**Comment by ganigeorgiev** [comment, marked as answer]
@emrahamet I think it is fine.

_The custom id length could be only an issue if the record is part of a multi-relation not-contains filter (eg. `someMultiRelField ~ "someId"`), but with the multi-match support in the upcoming v0.11 this no longer will be a concern._

2 small nitpicks:
- The empty path route is not required since you are handling the empty id case in the `view`.
- `NewNotFoundError` and `EnrichRecord` needs to be prefixed with the `github.com/pocketbase/pocketbase/apis` package.

**Reply by ganigeorgiev** [reply]
@emrahamet Because of the used `apis` package name and the missing prefix in `NewNotFoundError` and `EnrichRecord`, just to make sure that you are extending PocketBase correctly - you are NOT overwriting/adding new files in a cloned PocketBase repo, right?

The correct way to extend PocketBase is to use it as a library in your own Go program and to attach the above handlers in the `OnBeforeServe` hook (see https://pocketbase.io/docs/use-as-framework/ and https://pocketbase.io/docs/router/). You can also check the example in https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go.
_If you are already doing this, then you can ignore this comment._
